Xamarin.Forms.Core/Items/MarshalingObservableCollection.cs

```diff
@@ -0,0 +1,157 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+
+namespace Xamarin.Forms
+{
+	// Wraps a List which implements INotifyCollectionChanged (usually an ObservableCollection)
+	// and marshals all of the list modifications to the main thread. Modifications to the underlying 
+	// collection which are made off of the main thread remain invisible to consumers on the main thread
+	// until they have been processed by the main thread.
+
+	public class MarshalingObservableCollection : List<object>, INotifyCollectionChanged
```

Comment:
public ?

Reply:
Used by Platform.Android, so it needs to be public.